### ISSUE TITLE
Incorrect results when checking WasPressed/WasReleased/IsPressed

### ISSUE THREAD

**Issue by mrkake** [pbhogan/InControl#191]
I'm not sure where else to write this, but I got everything set up with InControl. Things work fine with one player using any device (keyboard, ps4 controller, xbox 360 controller are what I'm using). As soon as you add a 2nd player things get weird.

At first everything seemed totally fine, but the more rapidly people press buttons at the same time things get messed up and I'm not sure what is going on.

I put some breakpoints in my code and the InControl code, and already I noticed a few weird things:
- Occasionally pressing a button a single time causes the following changes to occur in 4 seperate frames:  "WasPressed == true", "WasReleased == true", ~10-15 or so frames in between "WasPressed == true" , "WasReleased == true". 

Another weird thing I then noticed, if I hold Action1 on whichever controller joins first, and then hold action 2 on a second controller, only one of the controllers sends a constant action1. The other controller stops sending action1 altogether, then as soon as you release Action1, controller 2 begins to send it again.

Another interesting thing is that none of the updates are marked as "IsPressed", only WasPressed/WasReleased constantly alternating between frames.

Any ideas would be appreciated, has anyone else ever reported something like this?

(basically, it kind of seems like the only device which is updating its input is the one marked ActiveDevice, even though in reality both devices are active).  And I am not even using InputManager.ActiveDevice in my code, I am looping through the full list of devices.

and, I'm not sure if this code will make much sense out of context, but my code which is getting the input is here, basically on start I build a dictionary of just the keys I care about. Each method that wants to recieve input is subscribed in an event and there is a dictionary of subscribers.

    public void Update(GameState currentState)
        {
            if (currentState == GameState.Initializing) return;

            Dictionary<InputControlType, InputSubscriber> subscriberLookup = GetSubscriberLookup(currentState);

            var keys = subscriberLookup.Keys.ToList();
            foreach (var inputControlType in keys)
            {
                var inputSubscriber = subscriberLookup[inputControlType];
                var devices = InputManager.Devices.ToList();
                foreach (var inputDevice in devices)
                {
                    InputControl control = inputDevice.GetControl(inputControlType);                    
                    if (control.WasPressed)
                    {
                        inputSubscriber.Dispatch(ButtonState.Pressed, inputDevice, control.Value);
                    }
                    else if (control.WasReleased)
                    {
                        inputSubscriber.Dispatch(ButtonState.Released, inputDevice, control.Value);
                    }
                    else if (control.IsPressed)
                    {
                        inputSubscriber.Dispatch(ButtonState.Held, inputDevice, control.Value);
                    }
                }
            }
        }

**Comment by mrkake** [comment]
I don't understand exactly what is happening, but it is something with InControl. If I replace the InControl code with Input.GetKey , everything works as expected, of course just with the keyboard (2 players on the same keyboard).  With the incontrol code, holding down two keys on the keyboard, only keypress from one of the keys makes it through (when there are two custom user profiles on the keyboard)

**Comment by mrkake** [comment]
Ok, just one more comment for now,  I changed the signature of the method above to public void public void Update(ulong ul, float fl)  ,  and I subscribed it to InputManager.OnUpdate ,  just because I was curious if this would help. When I do this, with a breakpoint I can see that all of the calls to GetControl are still there, but none of the buttons do anything and all of the ispressed/waspressed/etc. are always false. I'm not sure if this is relevant or not, or the purpose of subscribing to OnUpdate.

I also attempted to subscribe to ActiveDevice changed, which seemed promising at first, but the problem is exactly what I said above -- the active device does not change. If you hold down a button on Controller 1, then start holding a button on Control 2, the active device stays as Controller 1 until Controller 1 is released or another button pressed, but weirdly it also seems to give inaccurate results.

**Comment by pbhogan** [comment]
That's super weird behavior. I've not seen or heard of anything like that. I wonder if it is due to a device or something else. Are you on Windows or Mac? Do you have Use Fixed Update enabled on InControlManager by chance?

**Comment by mrkake** [comment]
I tried with fixed update and without fixed update and both are exhibiting the behavior. I don't think it's a device, and I'm seeing this on multiple computers, I'm working on the game with my roommate and the same problem occurs there (he is using a keyboard and xbox 360 controller). 

I tried looking in the incontrol code to see if there was anything obvious but not really, it seems pretty straightforward and its just using Input.GetKey at the end of the day so I don't really understand.

**Comment by pbhogan** [comment]
I'll do some experimenting and see if I can get it to happen. Can you provide a minimal example with steps to reproduce the weirdness?

**Comment by mrkake** [comment]
I will try to create a minimal example but we will see if it works.
If you have time, and you want, I can share my screen with you also.

**Comment by mrkake** [comment]
Wow, I am so sorry for all the trouble that I have given you lately with this, and with the earlier pull request. I created the whole sample project and discovered that there was no problem at all.

Then, I went back into my code and found the issue somewhere else, basically, in another class I had logic to try to "rate limit" holding down the button action.

The code looks something like this:

    if (TimeRecorder.HasTimeElapsed(button.Device, GameConfig.PaddleFrequency))
            {
                TimeRecorder.Record(button.Device);
                Paddle(player);
            }

but previously it looked like this:

   if (TimeRecorder.HasTimeElapsed(string.Format("paddle{0}", button.Device), GameConfig.PaddleFrequency))
            {
                TimeRecorder.Record(string.Format("paddle{0}", button.Device));
                Paddle(player);
            }

I don't know why for the life of me I thought that button.Device (which is type InputDevice) would convert to a string, actually I think it's a remainant from when a few days ago I was using the InputDevice.Name to track the device. I refactored the code to use the actual object, but I missed this string.Format. 

Man I can't believe that. I really appreciate your patience, and I'm going to buy InControl on the asset store now instead of just leeching off the github version.
